### PR TITLE
Misc form work and styling

### PR DESCRIPTION
**Pull request by Wcarey7**
- Delete customer modal. Confirmation modal now pops up when delete button is clicked. Page reloads after each delete, if the last row is deleted the previous page is loaded.

- Favicon image Located in the static folder. Image shows up on the tab in the web browser, the little icon next to the title. Gets rid of the 404 code for the favicon in the console.

- Removed debug toolbar extension. No longer needed, it flooded the console with many GET requests, and sometimes failed.

- user_profile folder. Moved the user_profile.html.j2 into its own folder under templates.

- Re-work of the error template pages for the 404 and 500 HTTP error codes. They previously weren't being rendered if called from certain routes. Text is now centered.

- Flashed messages (i.e. "You Are Now Logged In", "New Customer Added") now slide up and dismiss after 3 seconds.

- Switched to using the shorthand CSS method of grid-column and grid-row.

- Consistent use of single/double quotes and more descriptive css/html class names.

### CODE REVIEW

File: app/templates/auth/register.html.j2
Comment (jdstone) [inline app/templates/auth/register.html.j2]: I think this should be (I added a + sign as an allowed character):

`^\w+([\.-\\+]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$`

Gmail email addresses allow a plus sign, such as `jdstone1+twitter@gmail.com`.

I think the regex is correct, I tested it here: https://regex101.com

File: app/templates/base.html.j2
Comment (jdstone) [inline app/templates/base.html.j2]: Did you add this into branch?  It already exists in main.  It looks like you just copied and pasted it in, but I could be wrong.

File: app/templates/customer/new_customer.html.j2
Comment (jdstone) [inline app/templates/customer/new_customer.html.j2]: This regex doesn't match the one above in the other file.  Also, a + sign needs to be allowed as I previously mentioned.

Also, since this code appears in more than one location, maybe we should take it out and put it somewhere central?  That way, when you update the code once, the code is the same everywhere it's used.

**Review by jdstone — CHANGES_REQUESTED** [review]
After the changes are made, re-request my approval.  Don't forget to rebase!  Good practice is to rebase regularly throughout development on a branch other than the primary branch ('main').  And always rebase before merging a PR, because if you don't, you may end up with issues or conflicts.  Let me know if you have questions.

File: app/templates/base.html.j2
Comment (Wcarey7) [reply app/templates/base.html.j2]: It's from the previous branch I did, New-Customer-Modal-Form, I created this branch from it after I submitted the PR for New-Customer-Modal-Form. So What happened was on Saturday I went to rebase this branch but I decided to try something new and used the GitLens interactive editor in VSCode. I must've done something wrong because it duplicated all my commits, the code was fine though but I ended up undoing the commits. I chose to rebase with main but it looks like it rebased with New-Customer-Modal-Form.
Commit Graph

File: app/templates/auth/register.html.j2
Comment (Wcarey7) [reply app/templates/auth/register.html.j2]: Thanks! I just pushed the changes now.

File: app/templates/customer/new_customer.html.j2
Comment (Wcarey7) [reply app/templates/customer/new_customer.html.j2]: I added the plus sign, I couldn't tell what other part didn't match though.

I agree, its on my todo list to create a Javascript directory under the Static directory and move all the scripts there to help avoid that duplication. Just been contemplating how to organize the JS files.

1.  Have all the scripts in one file, such as Main.js
2. Organized by "function", so an inputValidation.js, deleteButton.js, Modals.js, etc.
3. Have them by page, newCustomer.js, login.js, editCustomer.js

File: app/templates/base.html.j2
Comment (jdstone) [reply app/templates/base.html.j2]: Got it.  When you rebase right before you merge the PR, are you marking most of the commits as "squash"? Or are you "picking" all of them?

File: app/templates/customer/new_customer.html.j2
Comment (jdstone) [reply app/templates/customer/new_customer.html.j2]: > I added the plus sign, I couldn't tell what other part didn't match though.

It's in the brackets: `[\.-]`  Just make sure they are the same.  Unless they are not supposed to match?  I would pick the correct one and copy/paste it to the other.

**EDIT:** I thought it was in the brackets, but I can't tell anymore.

>
> I agree, its on my todo list to create a Javascript directory under the Static directory and move all the scripts there to help avoid that duplication. Just been contemplating how to organize the JS files.
> 
> 1. Have all the scripts in one file, such as Main.js
> 2. Organized by "function", so an inputValidation.js, deleteButton.js, Modals.js, etc.
> 3. Have them by page, newCustomer.js, login.js, editCustomer.js

Yeah, we can talk about this and figure it out together.

**Review by jdstone — COMMENTED** [review]
Something is still funky with these regex expressions.  I could be wrong with what I showed in adding the + sign.  Just double check the regex expressions.

I'll approve the merge once the conflicts are resolved.

I think creating this branch off your other branch may not have been a good idea, I don't know.

File: app/templates/base.html.j2
Comment (Wcarey7) [reply app/templates/base.html.j2]: I only selected a couple to squash.

Comment (jdstone) [reply app/templates/base.html.j2]: We don't want all of these commits added to the history. It just clusters it up. For example, you have more than one favicon commit and multiple commits related to the customer modal. Consolidate those and any others you see into one commit. You can use reword to change the commit message to something more descriptive.

Do you want me to show you what I mean?

Comment (Wcarey7) [reply app/templates/base.html.j2]: Yeah, how do I reword it? I cant find an option to do it. The commits didn't get squashed correctly like I thought they would when I attempted the rebase.

Comment (jdstone) [reply app/templates/base.html.j2]: Let's go into it tonight. I don't want to do it too late. Call me between 5 and 6. I'll be unavailable between 3 and 5, but should be back home by 5 or so.

File: app/templates/customer/new_customer.html.j2
Comment (jdstone) [reply app/templates/customer/new_customer.html.j2]: This looks good.  They are all the same.

**Review by jdstone — CHANGES_REQUESTED** [review]
Let's rebase this together and get this completed.  When are you available to do this?  Let me know.

**Review by jdstone — APPROVED** [review]

**Comment by jdstone** [comment]
Old duplicate PR. Closing.

**Pull request closed without merging**
